test/powershell/Language/Parser/MethodInvocation.Tests.ps1

```diff
@@ -81,3 +81,110 @@ namespace MSFT_716893
         ([MSFT_716893.IInterface1]$proxy).BaseOperation(22) | Should -Be "3 - 22"
     }
 }
+
+Describe 'Generic Method invocation' -Tags 'CI' {
```

Comment:
A few other test suggestions:

- The big one for me is different kinds of error
    - `[array]::empty[notarealtype]()`
    - `[array]::empty[type]]()`
    - `[array]::empty[type, type]()`
    - `$object.Method[type,]()`
    - `$object.Method[]()`
    - `$object.Method[,]()`
    - `$object.Method[,type]()`
    - `$object.Method[type()`
    - `$object.Method[type)`
- Also tests around non-invoked methods (the method group info result):
    - `$object.Method[type]`
    - `$object.Method[type, type]`
- And something like `[Enumerable]::Select[List[int]](...)`

Reply:
> Also tests around non-invoked methods (the method group info result):
> * $object.Method[type]
> * $object.Method[type, type]

Hmm, that's a good point. The logic that retrieves method group info happens before (and separately to) the logic that actually resolves the method. I will double check this evening, but from what I recall, there isn't an easy way to pull the resolution logic into the method group result.

It might be doable, but I'm unsure if it is worth the effort in this instance. Currently I think the results from `$object.Method` and `$object.Method[...types...]` are identical, the resolution is only done when it's actually invoked at the moment. 

I'll look to add some more tests along those lines as well though, thanks!

Reply:
> Currently I think the results from $object.Method and $object.Method[...types...] are identical, the resolution is only done when it's actually invoked at the moment.

Yeah, I can see that making sense. The important part, I think, is that we capture what the expected behaviour is with tests.

Reply:
I've added test cases for all the variants you mentioned. I think they're all behaving in a pretty consistent way. Had to go in and figure out how to give some better runtime errors for cases where a method overload can't be found / the type(s) specified don't exist, and added some tests for those cases too.